Generator/doc/Generator/CGAL/point_generators_2.h

```diff
@@ -326,32 +326,98 @@ typedef const Point_2* pointer;
 
 */
 typedef const Point_2& reference;
+ /*!
+ Creates  an input iterator `g` generating points of type `Point_2` uniformly
+ distributed inside the triangle with vertices \f$ p, q \f$ and \f$ r \f$, i.e., \f$*g = \alpha p + \beta q + \gamma r \f$, for some
+ \f$ \alpha, \beta, \gamma \in [0, 1] \f$ and \f$ \alpha + \beta + \gamma = 1 \f$.
+ Two random numbers are needed from `rnd` for each point.
 
+ */
+ Random_points_in_triangle_2(Point_2& p, Point_2& q, Point_2& r, Random& rnd =
+ default_random);
 
+ /*!
+ Creates  an input iterator `g` generating points of type `Point_2` uniformly
+ distributed inside a triangle \f$t\f$ with vertices \f$ p, q \f$ and \f$ r \f$, i.e., \f$*g = \alpha p + \beta q + \gamma r \f$, for some
+ \f$ \alpha, \beta, \gamma \in [0, 1] \f$ and \f$ \alpha + \beta + \gamma = 1 \f$.
+ Two random numbers are needed from `rnd` for each point.
 
-/*!
-Creates  an input iterator `g` generating points of type `Point_2` uniformly
-distributed inside the triangle with vertices \f$ p, q \f$ and \f$ r \f$, i.e., \f$*g = \alpha p + \beta q + \gamma r \f$, for some
-\f$ \alpha, \beta, \gamma \in [0, 1] \f$ and \f$ \alpha + \beta + \gamma = 1 \f$.
-Two random numbers are needed from `rnd` for each point.
+ */
+ Random_points_in_triangle_2(Triangle_2& t, Random& rnd =
+ default_random);
 
-*/
-Random_points_in_triangle_2(Point_2& p, Point_2& q, Point_2& r, Random& rnd =
-get_default_random());
+ /// @}
+
+ }; /* end Random_points_in_triangle_2 */
+
+ /*!
+
+ The class `Random_points_on_triangle_mesh_2` is an input iterator creating points uniformly
+ distributed inside a Triangulation_2. The Triangulation_2 must have a face model refining `DelaunayMeshFaceBase_2`.
+ The sampled model is the union of the faces for which `DelaunayMeshFaceBase_2::is_in_domain()` returns `true`.
```

Comment:
inside a 2D domain represented by a 2D triangulation. `Triangulation` must be a \cgal triangulation with a face model of ....